src/Microsoft.MobileBlazorBindings/Elements/Handlers/PickerHandler.cs

```diff
@@ -0,0 +1,118 @@
+                {
+                    renderer.Dispatcher.InvokeAsync(() => renderer.DispatchEventAsync(SelectedItemChangedEventHandlerId, null, new ChangeEventArgs { Value = PickerControl.SelectedItem }));
+                }
+            };
+
+            ConfigureEvent(
+                eventName: "onselectedindexchanged",
+                setId: id => SelectedIndexChangedEventHandlerId = id,
+                clearId: id => {
+                    if (SelectedIndexChangedEventHandlerId == id)
+                    {
+                        SelectedIndexChangedEventHandlerId = 0;
+                    }
+                });
+
+            PickerControl.SelectedIndexChanged += (s, e) =>
+            {
+                if (SelectedItemChangedEventHandlerId != default)
+                {
+                    renderer.Dispatcher.InvokeAsync(() => renderer.DispatchEventAsync(SelectedIndexChangedEventHandlerId, null, new ChangeEventArgs { Value = PickerControl.SelectedIndex }));
+                }
+            };
+        }
+        public ulong SelectedItemChangedEventHandlerId { get; set; }
+        public ulong SelectedIndexChangedEventHandlerId { get; set; }
+        public XF.Picker PickerControl { get; }
+
+        public override void ApplyAttribute(ulong attributeEventHandlerId, string attributeName, object attributeValue, string attributeEventUpdatesAttributeName)
+        {
+            switch (attributeName)
+            {
+                case nameof(XF.Picker.CharacterSpacing):
+                    PickerControl.CharacterSpacing = AttributeHelper.StringToDouble((string)attributeValue, 0.0);
+                    break;
+                case nameof(XF.Picker.FontAttributes):
+                    PickerControl.FontAttributes = (XF.FontAttributes)AttributeHelper.GetInt(attributeValue);
+                    break;
+                case nameof(XF.Picker.FontFamily):
+                    PickerControl.FontFamily = (string)attributeValue;
+                    break;
+                case nameof(XF.Picker.FontSize):
+                    PickerControl.FontSize = AttributeHelper.StringToDouble((string)attributeValue, -1.00);
+                    break;
+                case nameof(XF.Picker.HorizontalTextAlignment):
+                    PickerControl.HorizontalTextAlignment = (XF.TextAlignment)AttributeHelper.GetInt(attributeValue);
+                    break;
+                case nameof(XF.Picker.ItemDisplayBinding):
+                    PickerControl.ItemDisplayBinding = new XF.Binding((string)attributeValue);
+                    break;
+                case nameof(XF.Picker.ItemsSource):
+                    var items = AttributeHelper.DelegateToObject<IList>(attributeValue);
+                    //IMPORTANT! don't set ItemsSource again if it is already set
+                    //Resetting ItemsSource will confuse the SelectedItem property and cause and infinite loop
+                    if (PickerControl.ItemsSource != items)
```

Comment:
That's interesting.
Could you help me to understand how to reproduce this issue? I've tried removing this `if`, but wasn't able to get anything wrong.

Reply:
I think I'll struggle to explain this one, and it took ages to debug.

Roughly, what happens is:
1. ItemsSource is set to a list
2. @bind-SelectedItem is used to bind the SelectedItem property on the picker to a property in the page
2. User selects an item e.g. `Item 3`(index 2) 
3. The setter on the proptery in the page is called to make the value be `Item 3`
4. After setter finished `StateHasChanged` is implicitly called
5. ApplyAttribute Runs in PickerHandler
6. XF.Picker.ItemsSource is set to the collection
7. When XF.Picker.ItemsSource is set, XF thinks of this as a reset so the SelectedIndex no longer makes sense
8. XF.Picker.SelectedIndex changes to -1
9. IndexChanging causes the setter on the proptery in the page is called to make the value be null
10. Go to step 4 and repeat infinitely with the wrong value.
11. Picker shows wrong value
12. App gets slow and buggy in seamingly unrelated ways.

Does that make sense? 

I'd love to hear your thoughts on a better solution if you can think of one, or if this causes any problems.

Reply:
Yeah I'm imagining that's exactly what's happening. Not all changes in Blazor cause a re-render, but if that property-set is executed again, I could see that it tricks Xamarin.Forms into thinking it's a brand new item source and re-sets the selection. It might be nice if it remembered it, but that's what we have.